lib/slather/coverage_service/html_output.rb

```diff
@@ -0,0 +1,214 @@
+require 'nokogiri'
+
+module Slather
+  module CoverageService
+    module HtmlOutput
+
+      def coverage_file_class
+        Slather::CoverageFile
+      end
+      private :coverage_file_class
+
+      def html_directory
+        "html"
+      end
+      private :html_directory
+
+      def post
+        create_html_reports(coverage_files)
+        generate_reports(@docs)
+      end
+
+      def create_html_reports(coverage_files)
+        create_index_html(coverage_files)
+        create_htmls_from_files(coverage_files)
+      end
+
+      def generate_reports(reports)
+        directory_path = html_directory
+        if output_directory
+          directory_path = File.join(output_directory, html_directory)
+        end
+
+        FileUtils.rm_rf(directory_path) if File.exists?(directory_path)
+        FileUtils.mkdir_p(directory_path)
+
+        reports.each do |name, doc|
+          html_file = File.join(directory_path, "#{name}.html")
+          File.write(html_file, doc.to_html)
+        end
+
+        # We could open 'index.html' automatically, although I don't know how to disable it for testing
```

Comment:
We could print a file URL, most terminals allow 'right click to open URL'. And it wouldn't cause trouble on CI servers.